PROCEDURES.md

```diff
@@ -0,0 +1,199 @@
+To release a `knative.dev` repository, these are the steps needed, in order:
+
+- ✅ [open PR check](#open-pr-check)
+- ✅ [build health check](#build-health-check)
+- ✅ [dependency check](#dependency-check)
+- ✅ [releasability check](#releasability-check)
+- ✅ [cut a branch](#cut-a-branch)
+- ✅ [automation check](#automation-check)
+- ✅ [release notes update](#release-notes-update)
+- ✅ [post release](#post-release)
+
+Most of the above steps are automated. In some noted exceptions, it might be
+necessary to perform some of them manually.cl
+
+For some repos, there might also be additional validations, or steps that need to be skipped. These exceptions are documented where they apply.
+
+## Open PR check
+Please ensure that there are no outstanding PRs for each repo before proceeding with each of the steps. For additional PRs merged into a repo during the release process, new checks need to be done in that repo and in the repos that depend on it.
+
+## Build health check
+tl;dr: check that all builds on `main` are passing.
+
+Check if the repository is in a good shape and the builds pass consistently. **This is required** because the [Prow job](#release-job) that builds the release artifacts will execute all the various tests (build, unit, e2e) and, if something goes wrong, the release process will probably need to restart from the beginning.
+
+For any problems in a specific repo, get in touch with the relevant WG leads to fix them.
+
+## Dependency check
+tl;dr: check that all the dependencies for the repo are up-to-date and aligned with the relase version.
+
+### Exceptions
+Repos that don't have dependencies naturally don't need a dependency check and this step can be skipped for those. Currently, `knative.dev/hack` is the only `knative.dev` repo that does not have any dependencies.
+
+### Aligning dependencies
+Each repo needs to be successfully updated to use the latest version of all shared dependencies **before** its release branch is cut.
+
+In order to align the `knative.dev` dependencies, `knative-sandbox/knobots` automation will run PRs like this [Upgrade to latest dependencies PR](https://github.com/knative/eventing/pull/4713) for each repo, executing the command `./hack/update-deps.sh --upgrade --release 0.20` and committing all the content. Note: `buoy check`, which is invoked in the script, will fail if the dependencies are not yet ready.
+
+- If there is no "Upgrade to latest dependencies" PR open, the update PR might already have been merged. If this is not the case, manually trigger the generation of this PR starting the [Knobots Auto Update Deps](https://github.com/knative-sandbox/knobots/actions/workflows/auto-update-deps.yaml) and wait for the PR to pop in the repo you need. Note that in the automation run you have to change the field `If true, send update PRs even for deps changes that don't change vendor. nUse this only for releases.` to **true**, because in some cases there are no code changes in the vendor.
```

Comment:
```suggestion
- If there is no "Upgrade to latest dependencies" PR open, the update PR might already have been merged. If this is not the case, manually trigger the generation of this PR starting the [Knobots Auto Update Deps](https://github.com/knative-sandbox/knobots/actions/workflows/auto-update-deps.yaml) and wait for the PR to pop in the repo you need. Note that in the automation run you have to change the field `If true, send update PRs even for deps changes that don't change vendor. Use this only for releases.` to **true**, because in some cases there are no code changes in the vendor.
```